torch/csrc/jit/fuser/compiler.cpp

```diff
@@ -263,7 +273,8 @@ std::shared_ptr<FusedKernel> compileKernel(
     const KernelSpec& spec,
     const ArgSpec& arg_spec,
     const std::vector<int64_t>& map_size,
-    const at::Device device) {
+    const at::Device device,
+    const at::ArrayRef<IValue> all_inputs) {
```

Comment:
This is a red flag. Why do we need exact inputs to compile a kernel that's supposed to be somewhat generic over the inputs we'll be getting? All information needed to identify which kernel to run should be part of the specs you get in here.

Reply:
This is where I wished I could get the lengths of the shape inputs in advance. It is those that I need.